core/tests/test_views.py

```diff
@@ -452,6 +452,31 @@ def test_create_community_view(client: Client, user: User) -> None:
     )
 
 
+def test_join_community_view(client: Client, user: User, community: Community) -> None:
+    client.force_login(user)
+    url = reverse("community-join", kwargs={"slug": community.slug})
+    response = client.post(url)
+    messages = list(get_messages(response.wsgi_request))
+    assert response.status_code == 302
+    assert any("You have joined the community!" in message.message for message in messages)
+
+    response = client.post(url)
+    messages = list(get_messages(response.wsgi_request))
+    assert any("You are already a member of this community." in message.message for message in messages)
+    assert response.status_code == 302
+
+    url = reverse("community-join", kwargs={"slug": "not_exist"})
+    response = client.post(url)
+    assert response.status_code == 404
+    assert "Community does not exist" in str(response.context)
+
+    client.alogout()
+    url = reverse("community-join", kwargs={"slug": community.slug})
+    response = client.post(url)
+    assert response.status_code == 302
+    assert "You have joined the community!" not in str(response.context)
```

Comment:
Yea, this is misleading. Having this in a separate test like "is_not_logged_in_is_unable_to_join_community" would be much clearer. I got confused. 
Also, a good test would be to show that this user was redirected to a login page or that there was an error message. This shows there was no success, which is hard to wrap my head around.